### PR TITLE
Add HSV color picker

### PR DESCRIPTION
**Pull request by qianlvable**

### CODE REVIEW

File: Paintroid/src/main/java/org/catrobat/paintroid/dialog/colorpicker/HSVColorPickerView.java [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+import android.graphics.RectF;
+import android.graphics.Shader;
+import android.util.AttributeSet;
+import android.view.MotionEvent;
+import android.view.View;
+
+import org.catrobat.paintroid.tools.implementation.BaseTool;
+
+/**
+ * TODO: document your custom view class.
```
Comment (X4mp) [inline Paintroid/src/main/java/org/catrobat/paintroid/dialog/colorpicker/HSVColorPickerView.java]: please remove/fix TODOs and comments

**Comment by X4mp** [comment]
ok to test

**Comment by X4mp** [comment]
please test this

**Comment by X4mp** [comment]
test this please

**Comment by X4mp** [comment]
2 tests failed within your implementation. Please fix them:
ColorDialogIntegrationTest.testIfRGBSeekBarsDoChangeColor with AssertionFailedError: No progress bars for ARGB :-( expected:<0> but was:<4>

ColorDialogIntegrationTest.testTabsAreSelectable with AssertionFailedError: Wrong tab count  expected:<3> but was:<2>

**Comment by qianlvable** [comment]
Thx for your patient.I will check it ASAP. [?]

2015-03-17 19:00 GMT+08:00 Roland Urbano <notifications@github.com>:

**Comment by X4mp** [comment]
@qianlvable can you update this pullrequest and fix the failing testcases? It would be great if we could merge the code for the HSV color picker.

**Comment by qianlvable** [comment]
Ok,I will check it out tonight.But I did`t seem any error detail.Can I run the test on my computer?

**Comment by X4mp** [comment]
great!
Yes you can run all tests on your local machine. Please check that all tests (at least the ones regarding your implementation) are passing successfully. I'll run the tests on our test environment as soon as you update the pull request.

**Comment by qianlvable** [comment]
Hi,Roland
I run test on my local machine,but I got many weird test fails seem not related to my change.
such as :
at org.catrobat.paintroid.test.integration.dialog.BrushPickerDialogIntegrationTest.testBrushPickerDialogOnBackPressed(BrushPickerDialogIntegrationTest.java:148)
on this line
assertEquals(paintStrokeWidth, newStrokeWidth);
Is there any chance my code base is not right?

**Comment by X4mp** [comment]
Hi,
we will run your test on our build system. You can see the testresults below.

**Comment by X4mp** [comment]
ok to test

**Comment by X4mp** [comment]
ok to test

**Comment by thmq** [comment]
ok to test

**Comment by AlxO** [comment]
There are some problems with the HSV Color picker:
 - with Android 6.0.1 and 5.1.1 the colorbox is only white
 - in the top left or bottom right corner there is the wrong color 
 - the white color is not shown in the box (only black at the top left corner)

screenshots:

**Comment by AlxO** [comment]
The problem with the new android versions is now fixed, but with API18 (also checked with the emulator) there is a SIGSEV (Fatal signal 11 (SIGSEGV) at 0x0000034c (code=1), thread 2582 (DrawingSurfaceT)). 

I do not really know if it is a problem of your implementation. I will look at it later.

**Comment by qianlvable** [comment]
Thanks for your patient！I will run on emulator to see if I can fix this problem too.

**Comment by AlxO** [comment]
Could you reproduce the bug?

**Comment by AlxO** [comment]
the issue was found. we will fix it ASAP.

thank you for your work.

**Comment by wslany** [comment]
@qianlvable Thanks a lot for contributing the HSV code! Are you already on our credits page, and if not, how should we mention you there? Kind Regards - Wolfgang

**Comment by qianlvable** [comment]
Hi,
Very sorry for replying so late. My VPN service is expired last month，so I had not access my Gmail for one month,as you know Google service is block in China .
I really excited to be list in credits page : )
Thanks

**Comment by wslany** [comment]
Hi, and great! So how can we refer to you on our credits page? What is your name? 
Thanks a lot, with kind regards, Wolfgang

**Comment by qianlvable** [comment]
My name is Jiaqi Ning. You can use it to refer to credits page ,thx

**Comment by wslany** [comment]
Great, I just added your name on http://developer.catrobat.org/credits --- Thanks a lot for your help! By the way, how accessible is Pocket Code in China at this time? Is it available via Google Play, or some alternative Chinese app stores? You can also contact me via email at wolfgang.slany@catrobat.org --- Thanks!

**Comment by qianlvable** [comment]
In China， most of android smart phones don`t have Google services，only geeks and developers hack their phone to install google service. You can find Pocket code in this WandouJia app store in this link ,this is a popular app store in my country : p
http://www.wandoujia.com/apps/org.catrobat.catroid <http://www.wandoujia.com/apps/org.catrobat.catroid>

**Comment by wslany** [comment]
Thanks! I see that they still have the 0.9.23 version. There's a newer, much better release on Google Play. Can we somehow upload it to the store ourselves? And which other popular stores are there in China?

**Pull request merged**